src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -82409,24 +82564,90 @@ static IC<T> CreateC<T>(T t)
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "iy.B(x)").WithLocation(29, 9));
         }
 
-        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/33347")]
+        [Fact, WorkItem(33347, "https://github.com/dotnet/roslyn/issues/33347")]
         public void NestedNullConditionalAccess()
         {
-            var source =
-@"class Node
+            var source = @"
+class Node
 {
     public Node? Next = null;
     void M(Node node) { }
     private static void Test(Node? node)
     {
         node?.Next?.Next?.M(node.Next);
     }
-}
-";
-            var comp = CreateCompilation(new[] { source }, options: WithNonNullTypesTrue());
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
             comp.VerifyDiagnostics();
         }
 
+        [Fact, WorkItem(31909, "https://github.com/dotnet/roslyn/issues/31909")]
+        public void NestedNullConditionalAccess2()
+        {
+            var source = @"
+public class C
+{
+    public C? f;
+    int? Test1(C? c) => c?.f.M(c.f.ToString()); // nested use of `c.f` is safe
+    int? Test2(C? c) => c.f.M(c.f.ToString());
+    int M(string s) => s.Length;
```

Comment:
>int M(string s) => s.Length; [](start = 4, length = 28)

Can this be removed?

Reply:
I've simplified `M` a bit, but otherwise left it. I need it to test access on `c.f` nested in expression with `c?.f`.